xwiki-platform-core/xwiki-platform-lesscss/xwiki-platform-lesscss-default/src/main/java/org/xwiki/lesscss/internal/cache/CacheKeyFactory.java

```diff
@@ -62,10 +74,25 @@ public String getCacheKey(LESSResourceReference lessResourceReference, SkinRefer
         String result = lessResource.length()  + CACHE_KEY_SEPARATOR + lessResource + CACHE_KEY_SEPARATOR
                      +  skin.length()          + CACHE_KEY_SEPARATOR + skin         + CACHE_KEY_SEPARATOR
                      +  colorTheme.length()    + CACHE_KEY_SEPARATOR + colorTheme;
-        
+
         if (withContext) {
             String xcontext = xcontextCacheKeyFactory.getCacheKey();
-            result += CACHE_KEY_SEPARATOR + xcontext.length() + CACHE_KEY_SEPARATOR + xcontext; 
+            result += CACHE_KEY_SEPARATOR + xcontext.length() + CACHE_KEY_SEPARATOR + xcontext;
+
+            /** Also take into account the request parameters, if any, except parameters which are already
+             * taken into account or that are irrelevant. */
+            XWikiContext context = xcontextProvider.get();
```

Comment:
An alternative to XWikiContext is to get the request from `org.xwiki.container.Container#getRequest()` and cast it into a `org.xwiki.container.servlet.ServletRequest` to have access to the `HttpServletRequest`.

Reply:
Thank Thomas, I just updated the code accordingly.